components/bitcoin-topics/Navigation.tsx

```diff
@@ -94,6 +106,57 @@ export function Navigation({
         )
     }, [expandedTopics])
 
+    // Update the useEffect to properly load completed topics
+    useEffect(() => {
+        const saved = localStorage.getItem("completedTopics")
+        if (saved) {
+            try {
+                const parsed = JSON.parse(saved)
+                // Convert the parsed object's keys into a Set
+                const completedSet = new Set(
+                    Object.keys(parsed).filter((key) => parsed[key])
+                )
+                setCompletedTopics(completedSet)
+            } catch (error) {
+                console.error("Error parsing completedTopics:", error)
+            }
+        }
+    }, [])
+
+    // Add listener for completion changes
+    useEffect(() => {
+        const handleCompletionChange = () => {
```

Comment:
this function is duplicated in `ln:111`, we can have the function outside the useEffect so the useEffects can consume it.

Also, I don't see the need of the first useEffect. This useEffect to handle `topicCompletionChanged` event is going to run on initial render, so the first one is unneeded.
Was there a specific need for the first useEffect?

Reply:
correct 
I combined these two effects since they both deal with the `handleCompletionChange` function